built-ins/draw-ROC-curve

```diff
@@ -0,0 +1,155 @@
+boxes:
+- id: anchor
+  inputs: {}
+  operationId: Anchor
+  parameters:
+    description: |-
+      Draws an ROC curve and computes the AUC
+      for a binary classifier prediction.
+      The two parameters are the true label (0 or 1)
+      and the predicted score from the model (between 0 and 1).
+
+      To avoid an overly detailed plot,
+      the curve is based on a sample of vertices.
+    parameters: >-
+      [{"kind":"vertex attribute (number)","id":"true
+      label","defaultValue":""},{"kind":"vertex
+      attribute (number)","id":"predicted
+      score","defaultValue":""},{"kind":"text","id":"sample
+      size","defaultValue":"1000"}]
+  parametricParameters: {}
+  x: 0
+  y: 0
+- id: Custom-plot_2
+  inputs:
+    table:
+      boxId: SQL1_5
+      id: table
+  operationId: Custom plot
+  parameters:
+    plot_code: |-
+      {
+        "layer": [{
+          "mark": "line",
+          "encoding": {
+            "x": {
+              "field": "fpr",
+              "title": "False positive rate",
+              "type": "quantitative"
+            },
+            "y": {
+              "field": "tpr",
+              "title": "True positive rate",
+              "type": "quantitative"
+            }
+          }
+        }, {
+          "mark": {
+            "type": "rule",
+            "color": "lightgray",
+            "strokeDash": [8, 8]
+          },
+          "encoding": {
+            "x": { "datum": 0 },
+            "y": { "datum": 0 },
+            "x2": { "datum": 1 },
+            "y2": { "datum": 1 }
+          }
+        }]
+      }
+  parametricParameters: {}
+  x: 700
+  y: 150
+- id: SQL1_4
+  inputs:
+    input:
+      boxId: input-input
+      id: input
+  operationId: SQL1
+  parameters:
+    persist: 'yes'
+    summary: Rename and filter
+  parametricParameters:
+    sql: |-
+      select
+        ${`true label`} as label,
+        ${`predicted score`} as score
+      from vertices
+      where isnotnull(${`true label`})
+        and isnotnull(${`predicted score`})
+      limit ${`sample size`}
+  x: 250
+  y: 250
+- id: input-input
+  inputs: {}
+  operationId: Input
+  parameters:
+    name: input
+  parametricParameters: {}
+  x: 50
+  y: 250
+- id: SQL1_5
+  inputs:
+    input:
+      boxId: SQL1_4
+      id: table
+  operationId: SQL1
+  parameters:
+    persist: 'no'
+    sql: |-
+      select
+      label, score,
+      sum(label) over (
+        order by score desc rows between
```

Comment:
I only found a small issue here: If it's possible that the classifier assigns the same score for multiple items, for that specific score we will get a lot of different `(fpr, tpr)` pairs. Only the last one (order by score desc) is really meaningful.

On the chart it can create weird blobs I think.

One possible solution is to add a second sql to pick the last value from each score-group, but it only works if we can assume that the order of the records remains the same between the first and the second sql. Or we can pre-aggregate by score `(num_1, num_0)` and the then your SQL will work even for repeated score values.

Reply:
> I only found a small issue here: If it's possible that the classifier assigns the same score for multiple items, for that specific score we will get a lot of different `(fpr, tpr)` pairs. Only the last one (order by score desc) is really meaningful.
> 
> On the chart it can create weird blobs I think.

Nope. Looks fine:

![image](https://user-images.githubusercontent.com/1268018/131658023-a6c30779-2167-443c-9721-593c4d9f2e5b.png)